database-commons/src/main/java/io/cdap/plugin/db/batch/source/AbstractDBSource.java

```diff
@@ -156,6 +156,10 @@ private Schema loadSchemaFromDB(Class<? extends Driver> driverClass)
     try (Connection connection = DriverManager.getConnection(connectionString, connectionProperties)) {
       executeInitQueries(connection, sourceConfig.getInitQueries());
       return loadSchemaFromDB(connection, sourceConfig.importQuery);
+
+    } catch (SQLException e) {
+      LOG.error("SQLException while performing getSchema", e);
+      throw new SQLException(e.getMessage(), e.getSQLState(), e.getErrorCode());
```

Comment:
why do we need new exception instance?

Reply:
db2 driver defines its own subclass of SQLException. This causing ClassDefNotFound exception when e.getMessage() called out of plugin classloader. Wrapping exception to plain SQLException helps to avoid ClassDefNotFound.